test/e2e/upgrade_test.go

```diff
@@ -0,0 +1,131 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	nc "github.com/openshift/windows-machine-config-operator/pkg/controller/windowsmachine/nodeconfig"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"log"
+)
+
+func upgradeTestSuite(t *testing.T) {
+
+	t.Run("Node deletion during upgrade", func(t *testing.T) { testUpgradeVersion(t) })
```

Comment:
This name leads me to believe that this is testing if a user deletes a node during the upgrade process